client/js/js_utils.js

```diff
@@ -26,181 +26,97 @@ function addLoadEvent(func) {
     }
   };
-  xmlhttp.open('GET', url, true);
-  xmlhttp.setRequestHeader('Fiware-Service', 'tourguide');
-  xmlhttp.send();
-}
 
+  xhr.onerror = function(e) {
+    on_failure_callback(this.response);
+  };
 
-function delete_ajax_petition(url, on_success_callback, on_failure_callback) {
-  var xmlhttp;
+  xhr.open('GET', url, true);
+  xhr.setRequestHeader('Fiware-Service', 'tourguide');
+  xhr.send();
+}
 
-  if (window.XMLHttpRequest) {
-    // code for IE7+, Firefox, Chrome, Opera, Safari
-    xmlhttp = new XMLHttpRequest();
-  }
-  else {
-    // code for IE6, IE5
-    xmlhttp = new ActiveXObject('Microsoft.XMLHTTP');
-  }
 
-  xmlhttp.onreadystatechange = function() {
-    if (xmlhttp.readyState == XMLHttpRequest.DONE) {
-      console.log(xmlhttp.responseText);
-      if (xmlhttp.status == 204) {
-        on_success_callback(xmlhttp.responseText);
-      }
-      else if (xmlhttp.status == 404) {
-        try {
-          on_failure_callback();
-        }
-        catch (err) {
-          on_failure_callback(xmlhttp.responseText);
-        }
+function delete_ajax_petition(url, on_success_callback, on_failure_callback) {
+  var xhr = new XMLHttpRequest();
 
-      }
-      else {
-        try {
-          on_failure_callback();
-        }
-        catch (err) {
-          on_failure_callback(xmlhttp.responseText);
-        }
-      }
+  xhr.onload = function(e) {
+    if (204 == this.status) {
+      on_success_callback(this.response);
+    }
+    else {
+      on_failure_callback(this.response);
     }
   };
-  xmlhttp.open('DELETE', url, true);
-  xmlhttp.setRequestHeader('Fiware-Service', 'tourguide');
-  xmlhttp.send();
-}
 
+  xhr.onerror = function(e) {
+    on_failure_callback(this.response);
+  };
+
+  xhr.open('DELETE', url, true);
+  xhr.setRequestHeader('Fiware-Service', 'tourguide');
+  xhr.send();
+}
 
 
 function post_ajax_petition(url, on_success_callback, on_failure_callback, data) {
-  var xmlhttp;
+  var xhr = new XMLHttpRequest();
 
-  if (window.XMLHttpRequest) {
-    // code for IE7+, Firefox, Chrome, Opera, Safari
-    xmlhttp = new XMLHttpRequest();
-  }
-  else {
-    // code for IE6, IE5
-    xmlhttp = new ActiveXObject('Microsoft.XMLHTTP');
-  }
 
-  xmlhttp.onreadystatechange = function() {
-    if (xmlhttp.readyState == XMLHttpRequest.DONE) {
-      if (xmlhttp.status == 201) {
-        on_success_callback(xmlhttp.responseText);
-      }
-      else if (xmlhttp.status == 404) {
-        try {
-          on_failure_callback();
-        }
-          catch (err) {
-          on_failure_callback(xmlhttp.responseText);
-        }
-      }
-      else {
-        try {
-          on_failure_callback();
-        }
-        catch (err) {
-          on_failure_callback(xmlhttp.responseText);
-        }
-      }
+  xhr.onload = function(e) {
+    if (200 == this.status) {
+      on_success_callback(this.response);
+    }
+    else {
+      on_failure_callback(this.response);
     }
```

Comment:
does it make sense to pass the response when there has been failure? what is the point?